### PR TITLE
Update metadata and address Openshift issues

### LINKED ISSUE #32: /model/metadata returns empty source information
```
{
  "id": "DCGAN",
  "name": "DCGAN Image Completer Model",
  "description": "DCGAN Image completion model; trained using Tensorflow on CelebA dataset",
  "type": "image-completion",
  "source": null,
  "license": "MIT"
}
```

### LINKED ISSUE #33: /model/predict endpoint fails on Red Hat OpenShift due to permission issue
```

[2019-08-06 20:26:12,510] ERROR in app: Exception on /model/predict [POST]
--
  | Traceback (most recent call last):
  | File "/opt/conda/lib/python3.6/site-packages/flask/app.py", line 1832, in full_dispatch_request
  | rv = self.dispatch_request()
  | File "/opt/conda/lib/python3.6/site-packages/flask/app.py", line 1818, in dispatch_request
  | return self.view_functions[rule.endpoint](**req.view_args)
  | File "/opt/conda/lib/python3.6/site-packages/flask_restplus/api.py", line 319, in wrapper
  | resp = resource(*args, **kwargs)
  | File "/opt/conda/lib/python3.6/site-packages/flask/views.py", line 88, in view
  | return self.dispatch_request(*args, **kwargs)
  | File "/opt/conda/lib/python3.6/site-packages/flask_restplus/resource.py", line 44, in dispatch_request
  | resp = meth(*args, **kwargs)
  | File "/workspace/api/predict.py", line 70, in post
  | os.mkdir(input_directory)
  | PermissionError: [Errno 13] Permission denied: '/workspace/assets/input'

```

The file system permissions are set as follows:

```
$ pwd
/workspace/assets
$ ls -al
total 16
drwxr-xr-x. 1 root root  4096 Jul 18 22:30 .
drwxr-xr-x. 1 root root  4096 Jul 22 20:25 ..
-rwxr-xr-x. 1  501 staff  212 Oct 16  2018 ._checkpoint
drwxr-xr-x. 2  501 staff 4096 Oct 16  2018 checkpoint
```

### PR DESCRIPTION
**Pull request by SSaishruthi**
This PR closes #33 and closes #32

### CODE REVIEW

File: api/predict.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def post(self):
         input_directory = '/workspace/assets/input'
         if not os.path.exists(input_directory):
             os.mkdir(input_directory)
+            os.chmod(input_directory, 0o777)
```
Comment (xuhdev) [inline api/predict.py]: The mode can be set within the [os.mkdir](https://docs.python.org/3/library/os.html#os.mkdir) call.

Comment (SSaishruthi) [reply api/predict.py]: I was about to do that. In documentation, it is mentioned that some systems ignore the mask. 

```
On some systems, mode is ignored. Where it is used, the current umask value is first masked out. If bits other than the last 9 (i.e. the last 3 digits of the octal representation of the mode) are set, their meaning is platform-dependent. On some platforms, they are ignored and you should call chmod() explicitly to set them.
```

Related link: https://stackoverflow.com/questions/5231901/permission-problems-when-creating-a-dir-with-os-makedirs-in-python

I am not sure how it works in openshift. I preferred this approach

Comment (xuhdev) [reply api/predict.py]: That's a good point!

File: api/predict.py [hunk 2]
```diff
@@ -64,96 +65,103 @@ def post(self):
...
-            os.mkdir(input_directory)
-        # clear input directory
-        for file in glob.glob(input_directory + '/*'):
+        # testing
+        with tempfile.TemporaryDirectory() as input_directory:
+            print('new directory', input_directory)
+            # creating directory for storing input
+            # input_directory = '/workspace/assets/input'
+            if not os.path.exists(input_directory):
+                # os.chmod('/workspace', 0o777)
```
Comment (bdwyer2) [inline api/predict.py]: Are these command no longer needed?

Comment (ptitzler) [reply api/predict.py]: No. The directory is automatically created. Lines 70 - 82 serve no purpose anymore

File: api/predict.py [hunk 3]
```diff
@@ -64,96 +65,103 @@ def post(self):
 
         image_input_read = Image.open(args['file'])
         image_mask_type = args['mask_type']
-        # creating directory for storing input
-        input_directory = '/workspace/assets/input'
-        if not os.path.exists(input_directory):
-            os.mkdir(input_directory)
-        # clear input directory
-        for file in glob.glob(input_directory + '/*'):
+        # testing
```
Comment (ptitzler) [inline api/predict.py]: testing?

File: api/predict.py (hunk 2 above)
Comment (SSaishruthi) [reply api/predict.py]: 1. This does not solve Openshift errors. It is not letting me create any directory in docker workspace. Patrick suggested me to try the tempfile approach.
2. Tempfile approach works but it is throwing `gateway time out` error even though the model runs fine. This is something to do with Openshift which Patrick is trying to figure out.

On hold because of this.

File: api/predict_copy.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
+#
```
Comment (ptitzler) [inline api/predict_copy.py]: Why's this file included?

File: core/model_copy.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
+#
```
Comment (ptitzler) [inline core/model_copy.py]: Why is this file included?

Comment (SSaishruthi) [reply core/model_copy.py]: As I have mentioned in our conversation a day before, I have not cleaned the code and waiting to see if it works with Openshift. Will clean up the unwanted files once we finalize with the approach.

**Comment by ptitzler** [comment]
We also need to update `test.py` to verify that the expected `source` information is returned.

File: api/predict.py (hunk 2 above)
Comment (ptitzler) [reply api/predict.py]: Resolved by increasing the router timeout to 90 seconds.

File: api/predict.py [hunk 6]
```diff
@@ -64,96 +65,97 @@ def post(self):
...
         image_input_read = Image.open(args['file'])
         image_mask_type = args['mask_type']
-        # creating directory for storing input
-        input_directory = '/workspace/assets/input'
-        if not os.path.exists(input_directory):
-            os.mkdir(input_directory)
-        # clear input directory
-        for file in glob.glob(input_directory + '/*'):
+        with tempfile.TemporaryDirectory() as input_directory:
+            # creating directory for storing input
```
Comment (ptitzler) [inline api/predict.py]: You can remove lines 69 - 77 because `tempfile.TemporaryDirectory()` creates an empty directory

File: core/model.py [hunk 7]
```diff
@@ -73,12 +73,13 @@ def _predict(self, model_data):
...
             args["maskType"] = model_data["mask_type"]
+            args["outDir"] = model_data["output_dir"]
 
             dcgan.complete(args)
 
         # Read output directory
 
         assert(os.path.exists(args["outDir"]))
-        output_image_path = args["outDir"] + '/completed/*'
+        output_image_path = args["outDir"] + '/completed'
```
Comment (ptitzler) [inline core/model.py]: Safer to use `os.path.join(...)`

**Comment by SSaishruthi** [comment]
Can I merge the PR?
